lib/libzfs/libzfs_pool.c

```diff
@@ -3312,6 +3312,7 @@ zpool_reopen(zpool_handle_t *zhp)
 	return (zpool_standard_error(hdl, errno, msg));
 }
 
+#if defined(__sun__) || defined(__sun)
```

Comment:
A comment describing why this functionality isn't required for Linux would be very helpful to our future selves.